DependencyInjection/HttplugExtension.php

```diff
@@ -106,6 +107,19 @@ private function configureClients(ContainerBuilder $container, array $config)
                 $container->setAlias('httplug.client.default', 'httplug.client.'.$first);
             }
         }
+
+        $mockServiceId = 'httplug.client.mock';
+
+        if (!\class_exists(MockClient::class)) {
+            $container->removeDefinition('httplug.factory.mock');
```

Comment:
Instead of removing a definition when the class does not exist, could you move the relevant service definition to a dedicated file and load this file if the class is available?